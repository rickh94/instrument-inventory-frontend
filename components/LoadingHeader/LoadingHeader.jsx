import React from 'react'
import PropTypes from 'prop-types'

import { CircularProgress, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'

const useStyles = makeStyles({
  heading: {
    display: 'flex',
    flexDirection: 'row',
  },
  loader: {
    marginLeft: '1rem',
  },
})

const LoadingHeader = ({ isLoading, title }) => {
  const classes = useStyles()
  return (
    <div className={classes.heading} data-testid="loading-header">
      <Typography variant="h5" color="inherit">
        {title}
      </Typography>
      {isLoading && <CircularProgress className={classes.loader} />}
    </div>
  )
}

LoadingHeader.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
}

export default LoadingHeader
