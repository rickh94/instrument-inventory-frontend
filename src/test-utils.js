/* eslint-disable react/prop-types */
import React from 'react'
import { MuiThemeProvider, createMuiTheme, indigo } from '@material-ui/core'
import { HelpersContext, SchemaContext } from './contexts'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

const theme = createMuiTheme({
  palette: {
    primary: indigo,
    secondary: {
      main: '#0A81D1',
    },
  },
})

export const showAlert = jest.fn()
export const setSearchResults = jest.fn()
export const setFilterResults = jest.fn()
export const searchResults = jest.fn()
export const filterResults = jest.fn()

export const TestTheme = ({ children }) => (
  <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
)

export const TestHelpers = ({ children }) => (
  <HelpersContext.Provider
    value={{
      showAlert,
      setSearchResults,
      setFilterResults,
      searchResults,
      filterResults,
    }}
  >
    {children}
  </HelpersContext.Provider>
)

export const TestRouter = ({ children }) => <MemoryRouter>{children}</MemoryRouter>

export const TestSchema = ({ children, schema = { components: { schemas: {} } } }) => (
  <SchemaContext.Provider value={schema}>{children}</SchemaContext.Provider>
)

export const TestEverything = ({ children }) => (
  <TestRouter>
    <TestTheme>
      <TestHelpers>{children}</TestHelpers>
    </TestTheme>
  </TestRouter>
)

// eslint-disable-next-line no-undef
export const flushPromises = () => new Promise(setImmediate)

const customRender = (ui, options, schema = { components: { schemas: {} } }) =>
  render(<TestSchema schema={schema}>{ui}</TestSchema>, {
    wrapper: TestEverything,
    ...options,
  })

export * from '@testing-library/react'

export { customRender as render }
