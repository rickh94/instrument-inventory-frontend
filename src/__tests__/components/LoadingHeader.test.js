import React from 'react'
import LoadingHeader from '../../components/LoadingHeader'
import { render } from '../../test-utils'
import 'jest-dom/extend-expect'

jest.mock(
  '@material-ui/core/CircularProgress',
  () =>
    function MockLoader() {
      return <div id="mock-loader" />
    }
)

describe('<LoadingHeader />', () => {
  it('matches snapshot', () => {
    const { container } = render(<LoadingHeader isLoading={false} title="test" />)
    expect(container).toMatchSnapshot()
  })

  it('renders title from props', () => {
    const { getByText } = render(<LoadingHeader isLoading={false} title="test title" />)
    expect(getByText('test title')).toBeInTheDocument()
  })

  it('shows a loader if isLoading is true', () => {
    const { container } = render(<LoadingHeader isLoading={true} title="test title" />)
    expect(container.querySelector('div#mock-loader')).toBeInTheDocument()
  })

  it('does not show a loader if isLoading is false', () => {
    const { container } = render(<LoadingHeader isLoading={false} title="test title" />)
    expect(container.querySelector('div#mock-loader')).not.toBeInTheDocument()
  })
})
