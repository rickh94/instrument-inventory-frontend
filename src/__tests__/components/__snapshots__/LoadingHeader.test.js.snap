// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingHeader /> matches snapshot 1`] = `
<div>
  <div
    class="makeStyles-heading-1"
    data-testid="loading-header"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
    >
      test
    </h5>
  </div>
</div>
`;
